pkg/cluster/deploystorage_resources.go

```diff
@@ -209,7 +208,7 @@ func (m *manager) networkPrivateLinkService(installConfig *installconfig.Install
 			},
 			Name:     to.StringPtr(m.doc.OpenShiftCluster.Properties.InfraID + "-pls"),
 			Type:     to.StringPtr("Microsoft.Network/privateLinkServices"),
-			Location: &installConfig.Config.Azure.Region,
+			Location: to.StringPtr(azureRegion),
```

Comment:
Why not just `&azureRegion`? I think `to.StringPtr` should only be used for string literals.